prometheus/metrics.go

```diff
@@ -164,71 +152,48 @@ func getServiceHealth(api v1.API, namespace, servicename string, ports []int32)
 	return ret, err
 }
 
-func getServiceMetrics(api v1.API, q *ServiceMetricsQuery) Metrics {
-	clustername := config.Get().ExternalServices.Istio.IstioIdentityDomain
-	destService := fmt.Sprintf(`destination_service="%s.%s.%s"`, q.Service, q.Namespace, clustername)
-	srcService := fmt.Sprintf(`source_service="%s.%s.%s"`, q.Service, q.Namespace, clustername)
-	if q.Service == "unknown" {
-		destService = `destination_service="unknown"`
-		srcService = `source_service="unknown"`
-		if q.Namespace != "unknown" {
-			destService += fmt.Sprintf(`,source_service=~".*\\.%s\\.%s"`, q.Namespace, clustername)
-			srcService += fmt.Sprintf(`,destination_service=~".*\\.%s\\.%s"`, q.Namespace, clustername)
-		}
-	}
-	labelsIn, labelsOut, labelsErrorIn, labelsErrorOut := buildLabelStrings(destService, srcService, q.Version, q.IncludeIstio)
-	groupingIn := joinLabels(q.ByLabelsIn)
-	groupingOut := joinLabels(q.ByLabelsOut)
+func getMetrics(api v1.API, q *MetricsQuery) Metrics {
+	labelsIn, labelsOut, labelsErrorIn, labelsErrorOut := buildLabelStrings(q)
+	groupingIn := strings.Join(q.ByLabelsIn, ",")
+	groupingOut := strings.Join(q.ByLabelsOut, ",")
```

Comment:
Note here: I removed the specific code for "Unknown". I don't know yet what we should do here (and don't remember why exactly we had this piece of code)

Reply:
Was this part of the pentagon implementation??
Also, "unknown" for destination and service was to pick up metrics coming from the outside that did NOT go through ingress (i.e. a browser request directly into a service found in the mesh) - perhaps this was to get metrics for those edges.

Reply:
I don't think this had anything to do with the pentagon support, that did not add anything to the metrics code iirc.  I think this is the right thing to do here, remove the special-case "unknown" handling and when we find out what we need, add it back * with comments *.